src/evaluator.ts

```diff
@@ -228,6 +242,20 @@ export class Evaluator {
     return false;
   }
 
+  private async evaluateClauses_v2(
+    clauses: Clause[],
+    target: Target,
+  ): Promise<boolean> {
+    for (const clause of clauses) {
+      if (!(await this.evaluateClause(clause, target))) {
+        // first clause to false, short-circuit and exit with false
+        return false;
+      }
+    }
+    // all clauses have passed
+    return true;
```

Comment:
What would happen if no clauses had been set up? Would this maybe cause the target to appear to be in the group erroneously?

Reply:
that's a good point, we probably want that returning `false` if `clauses` is empty